vulcan/build/create-js-routes

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env sh
+
+set -e
+
+# The webpack build process depends on gem setup in order to build the routes, but it cannot run the bundle install
+# itself without conflicting with the ruby server, so it merely checks and waits before invoking.
+if [ -n "$RUN_NPM_INSTALL" ]; then
```

Comment:
Is... this a pattern we want to repeat?  I'm honestly not sure this is wise.  I feel that compiling routes into the bundle makes building more complicated, tests more complicated (those route functions have to get mocked), and still don't really guarantee much about the integration (the exchange, headers, response payload schemas aren't captured anyways). But it's not a blocker.

Reply:
Normally I am definitely in favor of code generation, I guess my concern is really that this code generation doesn't give a lot (just routes, nothing about endpoint schemas), and it comes with acost.  If it could generate _more_ useful things, I'd be more keen.  But, in retrospect it's probably just fine.

Reply:
What would you recommend as an alternative? Would you just bake in the paths in the methods that call the API?

Reply:
Yeah, I would.  If we wanted, the paths could be considdated in a global or group of functions (similar to how it is, but just baked into the code).

In reality, if the api path changes, it's highly likely other aspects of the api change too, so the code is likely in need of udpate.  The main advantage is that tests themselves can be written without worrying too much about mocks, and you're not forced to codegen just to run js tests.